docroot/modules/custom/layout_builder_custom/layout_builder_custom.module

```diff
@@ -1116,22 +1134,35 @@ function _layout_builder_custom_add_lbs_heading(&$form) {
  */
 function layout_builder_custom_preprocess_block(&$variables) {
   // Unset grid_columns classes if display as list.
-  if (isset($variables['elements']['#attributes']['class'])) {
-    if (is_string($variables['elements']['#attributes']['class'])) {
-      $variables['elements']['#attributes']['class'] = explode(' ', $variables['elements']['#attributes']['class']);
+  if (isset($variables['attributes']['class'])) {
+    if (is_string($variables['attributes']['class'])) {
+      $variables['attributes']['class'] = explode(' ', $variables['attributes']['class']);
     }
-    $is_list = array_search('inner--list', $variables['elements']['#attributes']['class']);
-    if ($is_list) {
-      foreach ($variables['elements']['#attributes']['class'] as $key => $style) {
-        if (str_starts_with($style, 'grid--')) {
-          unset($variables['elements']['#attributes']['class'][$key]);
-        }
+
+    // If this is a views block...
+    if ($variables['base_plugin_id'] === 'views_block') {
+      // Get a list of styles that will be displayed on the view instead of the
+      // block and remove them.
+      $filtered_styles = LayoutBuilderStylesHelper::filterStyles($variables['attributes']['class'], [
+        'grid',
+        'inner',
+        'list',
+      ]);
+      LayoutBuilderStylesHelper::removeStylesFromAttributes($variables['attributes'], $filtered_styles);
+    }
+    else {
+      // Otherwise, remove grid classes if this is a list.
+      if (in_array('inner--list', $variables['attributes']['class'])) {
```

Comment:
should this be list-container now?

Reply:
`list-container--list`

Reply:
I'd like to talk about this a bit more before we roll this change out any further.